### ISSUE TITLE
merged `where.not` query in scope for has_many returns no relations

### ISSUE THREAD

**Issue by gnfisher** [rspec/rspec-core#2309]
First issue raised, excuse me if I do a poor job of this.

This is the query in question, which works in rails console and elsewhere fine but does not behave as expected in my rspec test suite.

```
  has_many :contributorships, dependent: :destroy
  has_many :contributors, source: :user, through: :contributorships, foreign_key: :user_id
  
  # Below is the actual query that misbehaves
  has_many :approved_contributors, -> { where(contributorships: { status: 3 }).merge(Contributorship.where.not(referral_type: 'creator')) },
            through: :contributorships,
            class_name: 'User',
            source: :user
```

It behaves as expected when the merged query is a simple `where` clause and not a `where.not`, e.g. `where(contributorships: { status: 3 }).merge(Contributorship.where(referral_type: 'creator'))`

Here is my (poorly written) test suite around this:

```
context '#approved_contributors' do
    let!(:idea_proof_point) { FactoryGirl.create(:idea_proof_point) }
    let!(:unapproved_contributor) { FactoryGirl.create(:user) }
		let!(:approved_contributor) { FactoryGirl.create(:user) }
		let!(:creator) { FactoryGirl.create(:user) }
		let!(:approved_contributorship) { FactoryGirl.create(:contributorship, idea_proof_point: idea_proof_point, user: approved_contributor, status: 3) }
		let!(:unapproved_contributorship) { FactoryGirl.create(:contributorship, idea_proof_point: idea_proof_point, user: unapproved_contributor, status: 0) }
		let!(:creator_contributorship) { FactoryGirl.create(:contributorship, idea_proof_point: idea_proof_point, user: creator, status: 3, referral_type: 'creator') }

    it 'returns approved contributors' do
    	expect(idea_proof_point.approved_contributors).to include(approved_contributor)
    end

    it 'does not return unapproved contributors' do
      expect(idea_proof_point.approved_contributors).not_to include(unapproved_contributor)
    end

		it 'does not return creator of idea' do
			expect(idea_proof_point.approved_contributors).not_to include(creator)
		end
  end
```

Basically it looks to me like merging a `where.not` query in a scope on a has_many through relationship breaks and returns no records. Possibly this is factory girl issue as well? Please let me know how I can be of better assistance. :)

**Comment by myronmarston** [comment]
Any issues with `where.not` should be reported to rails, not to RSpec.  RSpec has nothing to do with that rails feature (or any rails feature).
